src/main/java/org/opensearch/plugin/insights/core/reader/LocalIndexReader.java

```diff
@@ -104,7 +116,7 @@ public List<SearchQueryRecord> read(final String from, final String to, String i
         while (curr.isBefore(end.plusDays(1).toLocalDate().atStartOfDay(end.getZone()))) {
             String indexName = buildLocalIndexName(curr);
             SearchRequest searchRequest = new SearchRequest(indexName);
-            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().size(1000);
```

Comment:
Might want to add a constant so this does not get lost in the code in case we want to change this in the future